allensdk/brain_observatory/behavior/behavior_ophys_session.py

```diff
@@ -463,20 +463,22 @@ def _get_roi_masks_by_cell_roi_id(self, cell_roi_ids=None):
         for ii, (_, row) in enumerate(table.iterrows()):
             output[ii, :, :] = _translate_roi_mask(row["image_mask"], int(row["y"]), int(row["x"]))
 
-        pixel_size_um = self.api.get_surface_2p_pixel_size_um()
-        spacing_mm = (pixel_size_um / 1000., pixel_size_um / 1000.)
-        unit = 'mm'
+        # Pixel spacing and units of mask image will match either the
+        # max or avg projection image of 2P movie.
+        max_projection_image = self.get_max_projection()
+        pixel_spacing = max_projection_image.spacing
+        unit = max_projection_image.unit
 
         return xr.DataArray(
             data=output,
             dims=("cell_roi_id", "row", "column"),
             coords={
                 "cell_roi_id": cell_roi_ids,
-                "row": np.arange(full_image_shape[0]) * spacing_mm[0],
-                "column": np.arange(full_image_shape[1]) * spacing_mm[1]
+                "row": np.arange(full_image_shape[0]) * pixel_spacing[0],
```

Comment:
This seems to go all the way back to SimpleITK. If I look [here](https://simpleitk.readthedocs.io/en/master/fundamentalConcepts.html) I see that `spacing[0]` is column and `spacing[1]` is row.
![image](https://user-images.githubusercontent.com/32312979/91886077-a5f9e780-ec3d-11ea-9465-5901c42daefd.png)

Reply:
Good catch, it looks like the previous implementation was doing the wrong thing and also forgot to account for the `pixel_spacing / 2` start point of coordinates.